model/vm_host.rb

```diff
@@ -72,17 +72,16 @@ def ip6_reserved_network(prefix = 79)
 
   # Generate a random network that is a slice of the host's network
   # for delegation to a VM.
-  def ip6_random_vm_network(prefix = 79)
-    subnet_bits = prefix - host_prefix
```

Comment:
Do we always want fifteen bits of entropy, or perhaps use fewer bits, e.g. 12 to keep the customer subnet size the same?

I kind of think the latter would be better...

I also am a bit confused as to how this code matches the commit message. It makes reference to maintaining a /80 for customers, but it looks like here the prefix would always be fifteen bits smaller than the host prefix, so how is that maintained?

Reply:
I'm having trouble getting this out. My problem is, if the provided 2 hosts have, let's say "2a01:4f8:10a:128b::/64" as the subnet network address. I want to configure them in this way;
1. First host: "2a01:4f8:10a:128b:1::/68"
2. Second host: "2a01:4f8:10a:128b:2::/68"

The above is not really possible with /68 subnet. Because, the network address of the /68 subnet will be the same as its /64 parent. The next hextet start at /80 which we give to the customer. Here in this PR, I tried to find a solution but failed. If we start the host with /80, we cannot allocate /80 to the customers. Is it important to be able to allocate? I'm not sure, because even if we allocate /80, today, they get /128 in their VM because of the dnsmasq DHCP limitations. (prefix delegation :/ )

Any suggestions?

Reply:
Can it be divided in half? Instead of using ::1 and ::2, you'd want more like `128b::2` and `128c::2` or something of the sort (possibly adding more bits), rather mixing things together. It seems like doing things with overlaps like that would also make routing really hard. Having little pinholes go to one host or another is somewhat anti-routing (which likes prefixes the most)

One thing I find puzzling and never resolved is how Comcast gives my laptop a /64: I presumed this was prefix delegation but now I'm not sure given our experience earlier. I never did anything special either. It was this ignorance that made me think it would have come together more naturally than things did. Perhaps I should break open tcpdump again and try to resolve what my client software is doing and the router is saying to me.

Reply:
Alright, I think I figured this out. So, we would divide it to 16 for /68. The child subnets would look like;
1. 2a01:4f8:10a:128b:0000::2/68
2. 2a01:4f8:10a:128b:1000::2/68
3. 2a01:4f8:10a:128b:2000::2/68
4. 2a01:4f8:10a:128b:3000::2/68
5. ....
16. 2a01:4f8:10a:128b:f000::2/68

This way, we would still be able to allocate /80 that belongs to that specific prefix in that specific host. Of course, the prefix will be set manually in case we get hosts behind the same /64 prefix.

Reply:
That looks pretty good. Maybe back it off to a size where we can fit 42 computers, so, steal six bits. A standard rack is 42U, is why. 

And indeed comcast uses slaac in addition to a dhcp /128, and after doing it I remember having done this before and concluding we had nothing to do until https://datatracker.ietf.org/doc/draft-mishra-6man-variable-slaac/ gets anywhere (at present, it don't look good)